test/jest.browser.setup.ts

```diff
@@ -11,7 +12,20 @@ beforeAll(async () => {
     serverOptions: {
       webpackConfig,
     },
-  })
+  }).then(
+    // NOTE: when upstream patch is fixed, remove this last promise handler
+    (browser) => {
+      patchServerConnectionInfo(
```

Comment:
Here is where we mutate `server` (see previous comment on `patched/PageWithPreviewServer.ts`) to fix `server.connectionInfo.url`.